File: apps/gustav/lib/features/onboarding/domain/household.dart
import 'weekday.dart';

enum HousingType { apartment, houseWithGarden }

enum Surroundings { city, suburb, countryside }

enum Experience { firstTimeOwner, experienced }

const _germanByHousingType = {
  HousingType.apartment: 'wohnung',
  HousingType.houseWithGarden: 'haus_garten',
};

const _germanBySurroundings = {
  Surroundings.city: 'stadt',
  Surroundings.suburb: 'vorort',
  Surroundings.countryside: 'land',
};

const _germanByExperience = {
  Experience.firstTimeOwner: 'ersthund',
  Experience.experienced: 'erfahren',
};

extension HousingTypeGerman on HousingType {
  String toGerman() => _germanByHousingType[this]!;
}

extension SurroundingsGerman on Surroundings {
  String toGerman() => _germanBySurroundings[this]!;
}

extension ExperienceGerman on Experience {
  String toGerman() => _germanByExperience[this]!;
}

/// Mirrors the `haushalt` table — one per owner, not per dog (multi-dog
/// households are backlog V2, see `docs/datenmodell.md`).
class Household {
  const Household({
    this.postalCode,
    required this.housingType,
    required this.surroundings,
    required this.experience,
    required this.weekdayTimeBudgetMinutes,
    required this.weekendTimeBudgetMinutes,
    this.trainingDays = const {},
    this.planningDay = Weekday.sunday,
    this.householdSize = 1,
    this.equipment = const [],
  });

  final String? postalCode;
  final HousingType housingType;
  final Surroundings surroundings;
  final Experience experience;
  final int weekdayTimeBudgetMinutes;
  final int weekendTimeBudgetMinutes;
  final Set<Weekday> trainingDays;
  final Weekday planningDay;
  final int householdSize;
  final List<String> equipment;
}

File: apps/gustav/src/features/onboarding/domain/household.ts
import type { Weekday } from './weekday';

export const HOUSING_TYPE_OPTIONS = [
  { value: 'apartment', label: 'Wohnung' },
  { value: 'houseWithGarden', label: 'Haus' },
] as const;
export type HousingType = (typeof HOUSING_TYPE_OPTIONS)[number]['value'];

export const SURROUNDINGS_OPTIONS = [
  { value: 'city', label: 'Stadt' },
  { value: 'suburb', label: 'Stadtrand' },
  { value: 'countryside', label: 'Land' },
] as const;
export type Surroundings = (typeof SURROUNDINGS_OPTIONS)[number]['value'];

export const EXPERIENCE_OPTIONS = [
  { value: 'firstTimeOwner', label: 'Erster Hund' },
  { value: 'experienced', label: 'Schon Erfahrung' },
] as const;
export type Experience = (typeof EXPERIENCE_OPTIONS)[number]['value'];

/** Mirrors the `household` table — one per owner, not per dog (multi-dog
 * households are backlog V2, see docs/datenmodell.md). */
export type Household = {
  postalCode: string | null;
  housingType: HousingType;
  surroundings: Surroundings;
  experience: Experience;
  weekdayTimeBudgetMinutes: number;
  weekendTimeBudgetMinutes: number;
  trainingDays: readonly Weekday[];
  planningDay: Weekday;
  householdSize: number;
  equipment: readonly string[];
};
